### ISSUE TITLE
Joining public link as logged in user doesn't keep channel visible 

### ISSUE THREAD

**Issue by PVince81** [nextcloud/spreed#4717]
## Steps to reproduce
1. Create two users admin and alice
2. Login as admin
3. Create a public room
4. Copy link
5. Open another browser (for a different session)
6. Login as alice
7. Open the link (which maybe admin sent by email or so)
8. Say hello in the chat
9. Log out
10. Log in again
11. Check conversations list

### Expected behaviour
Since I've joined and even posted as a regular user, I should be "fully joined"

### Actual behaviour
The conversation has disappeared from the list even though it was there before => user confused

## Talk app

**Talk app version:** 10.0.1

## Server configuration

**Nextcloud Version:** 20.0.0

## Notes

I was studying this behavior as part of https://github.com/nextcloud/spreed/pull/4706 because I think it should be changed there to allow "fully joined" mode. If we agree on the "expected behavior" we could also backport only the bits that make it work that way.

## Todos

- [ ] BLOCKED by [ability to manually join rooms](https://github.com/nextcloud/spreed/issues/4722)

**Comment by PVince81** [comment]
This happens because when a user joins with a public link they always use "USER_SELF_JOINED" mode, which works differently than a regular "USER" mode join. The former mode is rather important for when joining without being logged in, in which case we want to cleanup the session and participant status to make the guest user leave.

I think this shouldn't apply for logged in users.

**Comment by PVince81** [comment]
hmm I'm mixing up the "Participant::GUEST" and "Participant::USER_SELF_JOINED".
so probably this behavior was always on purpose ?

**Comment by nickvergessen** [comment]
Yeah this is currently the designed and wanted behaviour.
It ensures that password changes have to be re-entered by users that joined themselves.

From my POV this is a will not fix until we implement the button to "Join conversation" based on the listed-conversation option.

**Comment by PVince81** [comment]
Okay, I see. I didn't realize that logged-in users also need to enter the password (I'm still discovering the scope of these features).

I also see a related issue that a moderator won't be able to add the user who self-joined either: https://github.com/nextcloud/spreed/issues/3462

I've raised another issue to discuss and address the "don't join automatically but with a button": https://github.com/nextcloud/spreed/issues/4722

**Comment by nickvergessen** [comment]
Now done with the "Open conversations" feature
